resolwe_bio/processes/slamdunk/alleyoop_collapse.py

```diff
@@ -61,6 +86,41 @@ def run(self, inputs, outputs):
         collapsed_output = name + '_collapsed.txt'
         os.rename(name + '_collapsed.csv', collapsed_output)
 
+        # normalize to TPM
+        tcount_tpm = compute_tpm(collapsed_output)
+
+        # Map gene symbols to feature IDs
+        res = resdk.Resolwe()
+        CHUNK_SIZE = 1000
+        feature_dict = {}
+        out_columns = [
+            'gene_symbol',
+            'length',
+            'readsCPM',
+            'readsTPM',
+            'conversionRate',
+            'Tcontent',
+            'coverageOnTs',
+            'conversionsOnTs',
+            'readCount',
+            'tcReadCount',
+            'multimapCount',
+        ]
+
+        input_features = tcount_tpm.index.tolist()
+        features_sublists = [input_features[i:i + CHUNK_SIZE] for i in range(0, len(input_features), CHUNK_SIZE)]
+
+        for fsublist in features_sublists:
+            features = res.feature.filter(
+                source=inputs.source,
+                species=inputs.slamdunk.species,
```

Comment:
Perhaps you could save `species` to its own variable and use it throughout the `run()` function?